src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -389,9 +389,15 @@ namespace OpenMS
           t = (float)it.getRT();
           rt.push_back(t);
         }
+        
+        auto spectrum = it.getSpectrum();
+        auto im_data = spectrum.getIMData();
```

Comment:
Check if IM data exists and has same size as spectrum. Otherwise access will segfault in line 396

Reply:
I agree with you, but I'm confused... 

`getIMData` seems to return a `std::pair<Size, DriftTimeUnit>` so `im_data.first` is an `unsigned long` not a vector. 

From what I understood, for each (retention_time, mass) couple we should have an array of IM data not just one right ?

Reply:
I think he means: check if the float data array that is at position imdata.first in the vector of float data arrays exists and had the same length as the spectrum.

Reply:
I added code for one function that contains a check.